coap-gateway/service/auth.go

```diff
@@ -99,37 +98,37 @@ func verifyChain(chain []*x509.Certificate, capool *x509.CertPool) (string, erro
 		}
 	}
 	if !ekuHasClient {
-		return "", fmt.Errorf("not contains ExtKeyUsageClientAuth")
+		return fmt.Errorf("not contains ExtKeyUsageClientAuth")
 	}
 	if !ekuHasServer {
-		return "", fmt.Errorf("not contains ExtKeyUsageServerAuth")
+		return fmt.Errorf("not contains ExtKeyUsageServerAuth")
 	}
-	return coap.GetDeviceIDFromIdentityCertificate(certificate)
+	_, err = coap.GetDeviceIDFromIdentityCertificate(certificate)
+	if err != nil {
+		return fmt.Errorf("cannot get deviceID from common name: %w", err)
```

Comment:
the device ID not part of the certificate's common name

Reply:
Fixed